src/ethereum_spec_tools/evm_tools/fixture_loader.py

```diff
@@ -346,55 +360,57 @@ def json_to_withdrawals(self, raw: Any) -> Any:
 
         return self._module("fork_types").Withdrawal(*parameters)
 
-    def json_to_blocks(
+    def json_to_block(
         self,
-        json_blocks: Any,
-    ) -> Tuple[List[Any], List[Hash32], List[bytes]]:
+        json_block: Any,
+    ) -> Tuple[Any, Hash32, bytes, Optional[str]]:
         """Converts json block data to a block object"""
-        blocks = []
-        block_header_hashes = []
-        block_rlps = []
-
-        for json_block in json_blocks:
-            if "rlp" in json_block:
-                # Always decode from rlp
-                block_rlp = hex_to_bytes(json_block["rlp"])
-                block = rlp.decode_to(self.Block, block_rlp)
-                blocks.append(block)
-                block_header_hashes.append(rlp.rlp_hash(block.header))
-                block_rlps.append(block_rlp)
-                continue
-
-            header = self.json_to_header(json_block["blockHeader"])
-            transactions = tuple(
-                self.json_to_tx(tx) for tx in json_block["transactions"]
-            )
-            uncles = tuple(
-                self.json_to_header(uncle)
-                for uncle in json_block["uncleHeaders"]
-            )
-
-            parameters = [
-                header,
-                transactions,
-                uncles,
-            ]
+        block_exception = None
+        for key, value in json_block.items():
+            if key.startswith("expectException"):
+                block_exception = value
+                break
```

Comment:
Might it be better to make a `Mapping[str, List[Type[Exception]]` and handle it all at the top level, or are our exception types too vague?

Reply:
This is only for parsing if the fixture expects any exception. All exception handling is done at the topmost level in the `run_blockchain_st_test` function

Reply:
There is the bit here: https://github.com/ethereum/execution-specs/pull/767/files#diff-22ed9d4c129a5cb0772b573f1d75a7468e431a3f75131680d9a5e12c5e3e456fR381

Reply:
I have moved this up directly into the `run_blockchain_st_test` function